DifferentiationInterface/ext/DifferentiationInterfaceReactantExt/onearg.jl

```diff
@@ -0,0 +1,80 @@
+struct ReactantGradientPrep{SIG, XR, GR, CG, CG!, CVG, CVG!} <: DI.GradientPrep{SIG}
+    _sig::Val{SIG}
+    xr::XR
+    gr::GR
+    compiled_gradient::CG
+    compiled_gradient!::CG!
+    compiled_value_and_gradient::CVG
+    compiled_value_and_gradient!::CVG!
+end
+
+function DI.prepare_gradient_nokwarg(
+        strict::Val, f::F, rebackend::AutoReactant, x, contexts::Vararg{DI.Context, C}
+    ) where {F, C}
+    _sig = DI.signature(f, rebackend, x; strict)
+    backend = rebackend.mode
+    xr = to_reac(x)
```

Comment:
We shouldn't save anything as a prep argument if a reactant array, I would keep this as if reactant array then xr is nothing otherwise to_rarray(x)